Enhancement: Use p-textarea instead of p-text-input for strings in schemas

zhen0 opened the pull request:
Closes https://github.com/PrefectHQ/prefect/issues/8447

Gives a bigger input box to better handle larger strings

After:

<img width="1173" alt="image" src="https://user-images.githubusercontent.com/40272060/233756841-f8e90cc1-c3b4-4190-9697-0b1cf9955b49.png">

Before:

<img width="554" alt="image" src="https://user-images.githubusercontent.com/40272060/233756853-34194700-d988-424e-ad14-2edd01b42dde.png">

pleek91 commented:
@zhen0 im not sure about this change. This would apply to all block schemas. A lot of which have properties that are strings which I'm not sure benefit from this change. The forms will get really long.

zhen0 commented:
That's a good call out but I'm not sure it makes that much difference to form length:

<img width="579" alt="image" src="https://user-images.githubusercontent.com/40272060/233757867-f8045534-de94-4a44-a2d0-7953a3e3ff03.png">

<img width="748" alt="image" src="https://user-images.githubusercontent.com/40272060/233757886-cbc014ef-1240-4f59-bac4-54853b8885e4.png">

Checking that made me realize that the textarea does not handle passwords so well so I've added a check for that and added back text-input there. 

Happy to discuss further though if you feel strongly.
